### PR TITLE
BOOKKEEPER-880 Make LedgerHandle implement AutoCloseable

### PR DESCRIPTION
**Pull request by eolivelli**

### CODE REVIEW

File: bookkeeper-server/src/test/java/org/apache/bookkeeper/client/BookKeeperTest.java [hunk 1]
```diff
@@ -290,4 +290,16 @@ public void readComplete(int rc, LedgerHandle lh, Enumeration<LedgerEntry> seq,
 
         startBKCluster();
     }
+
+    @Test
```
Comment (rakeshadr) [inline bookkeeper-server/src/test/java/org/apache/bookkeeper/client/BookKeeperTest.java]: Its good practise to add time out in tests, @Test(timeout = 60000). Typo in test case name, testAutoCloseableBooKeeper.

File: bookkeeper-server/src/test/java/org/apache/bookkeeper/client/BookKeeperTest.java [hunk 2]
```diff
@@ -290,4 +290,16 @@ public void readComplete(int rc, LedgerHandle lh, Enumeration<LedgerEntry> seq,
...
+    public void testAutoclosableBooKeeper() throws Exception {
+        ClientConfiguration conf = new ClientConfiguration()
+            .setZkServers(zkUtil.getZooKeeperConnectString());
+        try (BookKeeper bkc = new BookKeeper(conf);
+                LedgerHandle lh = bkc.createLedger(digestType, "testPasswd".getBytes());){
+            for (int i = 0; i < 100; i++) {
+                lh.addEntry("foobar".getBytes());
+            }
+        }
```
Comment (rakeshadr) [inline bookkeeper-server/src/test/java/org/apache/bookkeeper/client/BookKeeperTest.java]: Please add assertion of ledger closure.

Assert.assertTrue("Ledger should be closed!", bkc.isClosed(ledgerId));

**Comment by rakeshadr** [comment]
Thanks Enrico for the patch. I've added few comments.

**Comment by merlimat** [comment]
Looks good. Can you also assert that the `bkc` instance is closed?

**Comment by merlimat** [comment]
:+1:

**Comment by rakeshadr** [comment]
Thanks Enrico. +1, lgtm.

**Pull request closed without merging**